README.md
# hand-polard

## Getting Started

Install dependencies,

```bash
$ npm i
```

Start the dev server,

```bash
$ npm start
```

Build documentation,

```bash
$ npm run docs:build
```

Build hand-polard via `father-build`,

```bash
$ npm run build
```

## 本地开发步骤

1. src/components 根据分类创建存放组件的文件夹 A
2. 文件 A 中需至少有 index.ts/index.js , index.md
3. md 格式：

```markdown
---
nav:
  title: 组件
group:
  title: 基础 (组件分类：基础，表单，附件，业务组件，其他，可自行归类或补充)
  path: /basic
---

## 组件名

## 代码演示

## API
```

4. src/index.ts 导出上方自定义组件
5. src/locale-language/zh-CN.ts, en-US.ts 维护多语言

## 本地开发注意事项

1. 本地 node 版本建议使用 v12.0.0 （个人使用的是 v14.16.0）以上，否则安装依赖时，dumi 内部无法兼容，导致项目启动报错
2. 组件内部如果需要使用 redux，建议使用 `window?.g_app?._store`
   > - 由于组件库的建立后行于项目，组件库内部用到的 dispatch 的相关 models 文件夹暂时只能使用项目中，如果组件库内部自行引入 react-redux 实现，恐项目用到了组件库的 models 文件，导致访问路径发生改变；反之则与项目耦合度过高
   > - 目前只有 lov 组件，search-area，custom-table（后二者共用一个）会用到
   > - dispatch 在这三个组件内的作用，仅实现数据的缓存，用于优化，不影响主体功能的使用

```js
// 全局入口
import { getDvaApp } from 'umi';
window.g_app = getDvaApp();

// 某js文件
const { dispatch, getState } = window?.g_app?._store;
```

或者

```js
// 某js文件
import { getDvaApp } from 'umi';
const { dispatch, getState } = getDvaApp()._store;
```

1. 涉及到 form 组件，一律改用 antd 4.x 中的 form，`Form.create()(component)`改为`WrappedForm(component)`[path: "components/wrapped-form"]

```js
export default WrappedForm(WrappedSearchArea);
```

4. 涉及到 connect 组件的使用，改用`WrapperConnect`[path: "components/custom-connect"]

```js
export default WrapperConnect(mapStateToProps)(CompatibleLov);
```

## 打包，发布

- 修改`.fatherrc.ts`文件中的`cjs,esm`
- 默认`rollup`打包模式，生成`dist`文件夹，该模式下体积小，但是 es6 语法
- 修改为`babel`模式时，会生成`lib`,`es`文件夹，该模式下体积大，同时需要修改`package.json`文件
  修改如下：

```js
{
  "main": "lib/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffect": false
}
// (改成这个貌似能实现一个假的按需加载，个人测试宿主项目中如果在babelrc中配置，经过依赖分析，也只是单独引用了某个组件)
{
  "main": "es/index.js",
  "module": "es/index.js",
  "typings": "es/index.d.ts",
  "sideEffect": false
}
```

且该模式下后续需要考虑做按需加载

记录踩坑：
babel 模式下打包，如果 tsx 文件中引入了.png 等，打包报错找不到图片，此时

1. 方案一，将该 tsx 文件改为 js 文件
2. 入口文件将 typings.d.ts 导入

```js
/// <reference path="../typings.d.ts" />
```

- 发布命令

```bash
$ npm run release
```

如果想发布指定版本号的版本，以 release 为例：

```bash
$ npm run build && npx standard-version -r v1.7.0 && npm publish
```

## 后续计划

1. 部分组件重构，减少代码量
2. 研究组件按需加载
3. 研究将组件打成一个压缩后的 js 文件

## 多语言

提供 LocaleProvider 组件，于项目入口出，调用该组件包裹子组件，并传入 locale 参数
如下是 antd 多语言与 hand-polard 多语言共有场景：

```tsx
// locale: {
//   locale: "zh_cn",
//   localeMap: {
//     zh_cn: {"common.key": "键",...},
//     en_us: {},
//   },
// }
//  组件库内部暂时仅支持zh_cn,en_us, 如项目有需求增加其他语言，可自行配置，并传入即可，localeMap同名配置以外界传入为主
<ConfigProvider locale={localeInfo[locale] || zhCN}>
  <IntlProvider onError={() => { }} locale={langTypes[locale]} messages={messages}>
    <InjectedWrapper>
      <LocaleProvider locale={{locale,localeMap}}>
        {basicLayout}
      </InjectedWrapper>
  </IntlProvider>
</ConfigProvider>
```

## 组件文档发布

```bash
$ npm run docs:build
$ docker build -t registry.cn-hangzhou.aliyuncs.com/ecf/polard-web:[年月日] .
$ docker push registry.cn-hangzhou.aliyuncs.com/ecf/polard-web:[年月日]
```


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.7.7](https://github.com/hand-rongjing/polard-component-library/compare/v1.7.6...v1.7.7) (2022-07-15)

### Bug 修复

- 增加新建多语言 - https://www.tapd.cn/34592457/bugtrace/bugs/view/1134592457001027598 ([47dba22](https://github.com/hand-rongjing/polard-component-library/commit/47dba22e083ae3226700be2364d97110c830b107))

### [1.7.6](https://github.com/hand-rongjing/polard-component-library/compare/v1.7.5...v1.7.6) (2022-07-13)

### 文档

- 更新日志 ([2cc88c9](https://github.com/hand-rongjing/polard-component-library/commit/2cc88c998cfcc2e9c6553baa519eb4b614c16688))

### Bug 修复

- 表格横向滚动条当数据<10 条时不出现-https://www.tapd.cn/34592457/bugtrace/bugs/view/1134592457001027502 ([5e38afe](https://github.com/hand-rongjing/polard-component-library/commit/5e38afe6fd2683d917d4f2448d59515379603fad))

### [1.7.5](https://github.com/hand-rongjing/polard-component-library/compare/v1.7.5-beta.0...v1.7.5) (2022-07-07)

### Bug 修复

- 凭证信息增加修改记录 - https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074121 ([68c1ab7](https://github.com/hand-rongjing/polard-component-library/commit/68c1ab77ff61f397cfb8f814ef67855701fe3979))
- 凭证信息增加修改记录 - https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074121 ([2d06bf9](https://github.com/hand-rongjing/polard-component-library/commit/2d06bf9ea603e951f84e4d27b034c4753fcd04df))
- 工作流预览增加财务共享节点-https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074172 ([ae5b3ee](https://github.com/hand-rongjing/polard-component-library/commit/ae5b3ee887642642766fe69663e6633448ba69b6))
- 工作流预览增加财务共享节点-https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074172 ([b3f1ef6](https://github.com/hand-rongjing/polard-component-library/commit/b3f1ef63e277bd6dbbbcec13d21c2ec19712bcf4))

### [1.7.5-beta.0](https://github.com/hand-rongjing/polard-component-library/compare/v1.7.4-beta.1...v1.7.5-beta.0) (2022-06-29)

### Bug 修复

- 凭证信息增加修改记录 - https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074121 ([6f9c926](https://github.com/hand-rongjing/polard-component-library/commit/6f9c926105f96ff9acd7c62ecf8adf4d676f999f))
- 核算工单增加工作流预览 - https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074735 ([7ea8ef0](https://github.com/hand-rongjing/polard-component-library/commit/7ea8ef0f0bc397c7d976a46bda0a0a6da7da6c18))

### [1.7.4-beta.1](https://github.com/hand-rongjing/polard-component-library/compare/v1.7.2-beta.1...v1.7.4-beta.1) (2022-06-23)

### 文档

- 增加组件路径日志 ([5ebd70b](https://github.com/hand-rongjing/polard-component-library/commit/5ebd70bc5392c72a34ce3fc8f7e3f044aa80cb4c))
- 增加组件路径日志 ([4745bc8](https://github.com/hand-rongjing/polard-component-library/commit/4745bc86782a36247c4c97792f9788d17311d893))

### Bug 修复

- col.render 存成\_ORG_RENDER_FN, 每次生成新的 render 函数来调用它, 触发 rc-table 的 Cell 的 render 【【需求收集登记】前端框架优化 - 【基础组】角色管理-分配数据权限后，编辑数据权限无效】https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074569 ([fe7ad28](https://github.com/hand-rongjing/polard-component-library/commit/fe7ad2813a10ee477aee57fa99ae2893ddcac8f3))
- EditTable 页码优化-https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074622 ([ffcf2f7](https://github.com/hand-rongjing/polard-component-library/commit/ffcf2f76de4691500397b0ef658fd5958caafbaf))
- trim console.log ([455128e](https://github.com/hand-rongjing/polard-component-library/commit/455128e3e026122965f3029ec6d2ff149a26068c))
- 使 dataSource 透传 【【需求收集登记】前端框架优化 - 【基础组】角色管理-分配数据权限后，编辑数据权限无效】https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074569 ([856be4a](https://github.com/hand-rongjing/polard-component-library/commit/856be4a2e8d469b511cf677f59c4d099a562223a))
- 生成 render 逻辑修改 ([13b2cf6](https://github.com/hand-rongjing/polard-component-library/commit/13b2cf681019ed522f39b76f034c85c2b3d2420d))
- 表格宽度不超出时，如果没有设置宽度则不显示滚动条 - https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074677 ([5a9e8bf](https://github.com/hand-rongjing/polard-component-library/commit/5a9e8bfa95c22d2341d0a889a81169db919cd1aa))

### [1.7.4-beta.0](https://github.com/hand-rongjing/polard-component-library/compare/v1.6.8...v1.7.4-beta.0) (2022-06-20)

### Bug 修复

- 【框架任务】【附件漏洞修复】- https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074187 ([75e4ca5](https://github.com/hand-rongjing/polard-component-library/commit/75e4ca536a8eb0e31439df369d8fb19c5ccbf7be))
- 【框架任务】【附件漏洞修复】- https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074187 ([46e082d](https://github.com/hand-rongjing/polard-component-library/commit/46e082dd38e7331f49e397704895bb21fc1f94c2))
- 【框架任务】【附件漏洞修复】- https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074187 ([e3f354a](https://github.com/hand-rongjing/polard-component-library/commit/e3f354a858ba1fd985bc14035995df3094dc0b4f))
- 【框架任务】【附件漏洞修复】- https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074187 ([c0a41f7](https://github.com/hand-rongjing/polard-component-library/commit/c0a41f7741231696237654ab08799c466603cd4d))
- 【框架任务】【附件漏洞修复】- https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074187 ([3a8f155](https://github.com/hand-rongjing/polard-component-library/commit/3a8f155fa552756b677dc97bd61e7d2897b1a795))
- col.render 存成\_ORG_RENDER_FN, 每次生成新的 render 函数来调用它, 触发 rc-table 的 Cell 的 render 【【需求收集登记】前端框架优化 - 【基础组】角色管理-分配数据权限后，编辑数据权限无效】https://www.tapd.cn/34592457/prong/tasks/view/1134592457001074569 ([fe7ad28](https://github.com/hand-rongjing/polard-component-library/commit/fe7ad2813a10ee477aee57fa99ae2893ddcac8f3))
- trim console.log ([455128e](https://github.com/hand-rongjing/polard-component-library/commit/455128e3e026122965f3029ec6d2ff149a26068c))
- 优化工作流预览样式 - https://www.tapd.cn/34592457/bugtrace/bugs/view?bug_id=1134592457001026932 ([46ef0f8](https://github.com/hand-rongjing/polard-component-library/commit/46ef0f8ddd5f634f08f979ba7af614f405bf8b76))

src/components/locale-language/en_US.ts
/*
 * @Author: binfeng.long@hand-china.com
 * @Date: 2021-10-22 15:33:39
 * @LastEditors: zong.wang01@hand-china.com
 * @LastEditTime: 2022-07-15 11:10:25
 * @Version: 1.0.0
 * @Description:
 * @Copyright: Copyright (c) 2021, Hand-RongJing
 */

export default {
  'common.please.enter': 'Please enter',
  'common.please.select': 'Please select',
  'common.all': 'All',
  'common.cannot.be.less.than': 'Cannot Be Less Than',
  'common.no.empty': '{name}cannot be empty',
  'common.no.matching.result': 'No matching result',
  'common.fold': 'fold',
  'common.expand': 'expand',
  'common.Illegal.digital': '{label} is an illegal number',
  'common.search': 'search',
  'common.reset': 'reset',
  'common.enabled': 'enabled',
  'common.disabled': 'disabled',
  'common.fill.in.multilingual': 'Fill In Multilingual',
  'common.no.more.than.limit': "Can't Exceed {number} Characters",
  'common.approval.history': 'Approval History',
  'common.print': 'print',
  'common.select.export.column': 'Please Select Export Column',
  'column.name': 'The column name',
  'common.columns.to.export': 'Select the columns to export',
  'common.export': 'export',
  'common.cancel': 'cancel',
  'common.export.as': 'Export to',
  'common.clear': 'clear',
  'common.save': 'save',
  'common.submit': 'submit',
  'common.approver': 'approver',
  'common.select.staff': 'Select the staff',
  'common.staff.code': 'Staff code',
  'common.staff.name': 'Staff name',
  'common.please.enter...': 'Please enter...',
  'common.checking.search.field.warning':
    'There are search fields that do not conform to the verification logic',
  'common.set.successfully': 'Set successfully',
  'common.delete.filter': 'Delete filter',
  'common.delete.confirm':
    'Are you sure to delete this condition? After deletion, you will no longer be able to use this filter',
  'common.delete.success': 'Delete successfully',
  'common.save.success': 'Save successfully',
  'common.default': 'default',
  'common.modified': 'modified',
  'common.set.default': 'Set the default',
  'common.rename': 'rename',
  'common.save.as': 'Save as',
  'common.clear.selected': 'Remove the selected',
  'common.set.key.before.save':
    'Please set a unique value to associate the filter with the current page',
  'common.rename.filter.criteria': 'Rename filter criteria',
  'common.new.filter': 'New filter',
  'common.set.filter.criteria.name': 'Set filter criteria name',
  'common.need.condition.name': 'Please enter a condition name',
  'common.name.of.warning': 'Please.enter.the.name.of.the.filter.criteria!',
  'common.name.spaces.warning': 'Filter criteria name cannot be all spaces!',
  'common.add.filter': 'Add filter criteria',
  'common.accounting.selected': 'Selected {count} item',
  'common.return': 'return',
  'common.edit': 'edit',
  'common.base.info': 'The basic information',
  'common.doc.preview.warning': 'This Document Cannot Be Previewed!',
  'common.doc.converting':
    'The Document Is Being Converted. Please Refresh The Page And Try Again Later.',
  'common.doc.conversion.failed':
    'The Document Conversion Failed, Unable To Preview!',
  'common.download': 'download',
  'common.enlarge': 'enlarge',
  'common.narrow': 'narrow',
  'common.rotate': 'rotate',
  'common.delete': 'delete',
  'common.upload.info': 'Click or drag file to this area to upload',
  'common.upload.support.ext': 'Support extension',
  'common.pic.size.warning': 'The Picture Size Does Not Meet The Requirements',
  'common.size.big.warning': 'Size Cannot Be Greater Than',
  'common.size.small.warning': 'Size Cannot Be Less Than',
  'common.attachment.size.limit':
    'Please select an attachment less than {size}{unit}',
  'common.attachment.size.lower':
    'Please select an attachment larger than {size}{unit}',
  'common.upload.failed.reason':
    'Upload Failed. The Current Attachment Type Format Setting Does Not Support This Kind Of Attachment Format Upload',
  'common.upload.success': 'Uploaded successfully',
  'common.upload.fail': 'Upload failed',
  'common.upload.not.allowed.delete':
    'The state does not allow attachments to be deleted',
  'common.operate.success': 'operate success',
  'common.upload.attachment': 'Attachment Uploaded',
  'common.preview': 'preview',
  'common.confirm.to.delete': 'Are you sure you want to delete it?',
  'common.compressed.package.preview': 'Compressed Package Preview',
  'common.attachment.upload.instructions': 'Attachment Upload Instructions',
  'common.upload.attachment.format': 'Upload Attachment Format',
  'common.size.of.attachments': 'Size Of Attachments That Can Be Uploaded',
  'common.attachment.any.size': 'Any size',
  'common.attachment.template': 'Attachment template',
  'common.click.to.upload': 'Click Upload Attachment',
  'common.upload.num': 'Upload Failed, Upload {num} Attachments At Most',
  'common.upload.size.range':
    'Upload Failed, File Size Should Be Within {range}',
  'common.description': 'Instructions',
  'common.voucher.date': 'Voucher Date',
  'common.currency': 'Currency',
  'common.original.currency.debit': 'Debit in original currency',
  'common.original.currency.credit': 'Original currency credit',
  'common.local.currency.debit': 'Debit in local currency',
  'common.local.currency.credit': 'Local currency credit',
  'common.large.class': 'Large Class',
  'common.name': '名称',
  'common.application.type': 'Application Type',
  'common.total.of.results': 'A Total Of {total} Results',
  'common.total': 'Total {total} Results',
  'common.recently.used': 'Recently Used',
  'common.line.number': 'Num',
  'common.tips': 'Tips',
  'common.reasons.for.submission': 'Reasons For Submission',
  'common.no.quality.inspection': 'No Quality Inspection',
  'common.under.quality.inspection': 'Under Quality Inspection',
  'common.state.finish': 'Finished',
  'common.close': 'Close',
  'common.entered': 'Entered',
  'common.in.volume': "It's in volume",
  'common.in.the.book': 'In the book',
  'common.entering': 'Entering',
  'common.being.destroyed': 'destruction.in.progress',
  'common.destroyed': 'destroyed',
  'efs.boxed': 'boxed',
  'efs.Warehoused': 'Warehoused',
  'common.editing': 'editing',
  'common.approving': 'approving',
  'common.withdraw': 'Withdraw',
  'common.auditing': 'In Audit',
  'common.approve.rejected': 'approve rejected',
  'common.approved': 'Reviewed',
  'common.audit.reject': 'Rejection Of Audit',
  'common.approve.pass': 'approve pass',
  'common.pending': 'Temporary hanging',
  'common.canceled': 'Cancel',
  'common.reviewed': 'Reviewed And Approved',
  'common.rejection.of.review': 'Rejection Of Review',
  'common.review.post': 'Review (post)',
  'common.recoil.submitted': 'Backflush Submission',
  'common.recoil.audit': 'Backflush Audit',
  'common.cancel.pending': 'Cancel hold',
  'common.pay': 'Pay',
  'common.payment.return': 'Refund',
  'common.refund': 'Refund',
  'acp.payment.reserved': 'Reserved',
  'common.copy.success': 'Replication Success',
  'common.attachments': 'Atch.',
  'base.duplicate.odd.number': 'Duplicate Odd Number',
  'common.amount': 'Amt.',
  'common.comment': 'Comment',
  'common.sequence': 'SN',
  'budget.control.method': 'Budget Control Method',
  'common.verification.info': 'Verification Information',
  'common.budget.verification.info': 'Budget Verification Information',
  'common.ok': 'OK',
  'common.create': 'New',
  'common.got.it': 'Got It',
  'common.budget.version': 'Budget version',
  'common.budget.structure': 'Budget structure',
  'common.budget.scenarios': 'Budget scenarios',
  'common.period': 'Period',
  'common.budget.item': 'Budget item',
  'common.expense.type': 'Expense type',
  'common.control.period': 'Phase control section',
  'common.company': 'Company',
  'common.company.group': 'Company group',
  'common.department': 'Department',
  'common.department.group': 'Department group',
  'common.responsibility.center': 'Responsibility Center',
  'common.responsibility.center.group': 'Responsibility Center Group',
  'common.user': 'User',
  'common.user.group': 'User group',
  'common.budget.item.type': 'Budget item type',
  'common.budget.item.group': 'Budget item group',
  'common.budget.amt': 'Amt.',
  'common.budget.rsv': 'Rsv.',
  'common.budget.usd': 'Usd.',
  'common.budget.avb': 'Avb.',
  'common.budget.schedule': 'Schedule',
  'common.budget.progress.info': 'Budget Progress Information',
  'exp.type.of.category': 'Document category',
  'common.a': 'individual',
  'common.batch.download': 'Bulk Download Of Attachments',
  'common.choose.download.format': 'Choose Download Format',
  'common.to.pdf.merge': 'To PDF Merge',
  'common.original.compression': 'Original Compression',
  'common.total.files': 'A Total Of {number} Files',
  'common.column.status': 'status',
  'base.successful': 'success',
  'base.fail': 'fail',
  'common.error.message': 'The error message',
  'common.status': 'Status: {status}',
  'common.spanned.file': 'Generating files',
  'common.import': 'Import',
  'common.choose.file': 'Select File',
  'base.strip': 'strip',
  'common.tpl.info.warning': 'Related Template Information Is Not Defined',
  'common.upload.excel': 'Please Upload Excel File',
  'common.click.tip': 'Click Import To Save The Successfully Imported Data',
  'common.upload.again': 'Please Modify The Error Data And Upload Again',
  'common.control.results.on.submission': 'Control Results On Submission',
  'common.please.enter.a.filter.name': 'Please enter a filter name',
  'common.filter.name.set.first': 'Please set the filter name first',
  'common.operation': 'operation',
  'common.info': 'Info',
  'common.table.search.warning':
    'You Currently Have Unsaved Changes. If You Perform A Search, The Data Will Be Reset.',
  'common.table.save.warning':
    'You currently have unsaved changes, data will be reset if jump.',
  'common.table.new.warning': 'There are unsaved data, please save it first!',
  'common.import.success': 'Import succeeded',
  'common.delete.warning': 'Are you sure to delete this row?',
  'common.show.total': 'show {range0}-{range1} of {total} item',
  'common.copy': 'copy',
  'common.cancel.fix': 'Cancel the fixed',
  'common.left.side.fix': 'Fixed on the left side',
  'common.right.side.fix': 'Fixed on the right side',
  'common.left.fix': 'Left fixed',
  'common.right.fix': 'Right fixed',
  'common.no.fix': 'Not fixed',
  'base.establish': 'establish',
  'common.download.import.template': 'Downloading and Importing a Template',
  'base.has.choose.count': 'Selected {count}',
  'base.count.options': 'Options:{count}',
  'common.download.error.data': 'Download Error Data',
  'common.yes': 'Yes',
  'common.no': 'No',
  'common.has.selected': 'Has selected',
  'common.not.selected': 'Not selected',
  'base.sync.check.subsidiary':
    'Check the subordinate company at the same time',
  'base.policy.verification.info': 'Policy Verification Information',
  'common.input.name.or.code': 'Please enter a code or name',
  'chooser.data.company': 'Choose the company',
  'common.to.configuration': 'To configure',
  'common.upload.max.num': 'Upload up to {filenum} files',
  'common.size.of.attachment':
    'The size of each attachment is {upperlimitsize}{unit}',
  'common.size.of.attachment.within':
    'The size of each attachment is within {lowerlimitsize}{unit} ~ {upperlimitsize}{unit}',
  'common.size.of.attachment.not.less.than':
    'The size of each attachment shall not be less than {lowerlimitsize}{unit}',
  'common.size.of.attachment.in':
    'The size of each attachment is within {upperlimitsize}{unit}',
  'full.screen': 'Full screen',
  'full.screen.exit': 'Exit full screen',
  'pre.page': 'The previous page',
  'next.page': 'The next page',
  'common.view.flow.chart': 'View flow chart',
  'common.view.flow.remark':
    'Note: The approval flow preview is displayed according to the current workflow, but will actually change according to configuration changes.',
  'approve.flow.preview': 'Approval Flow preview',
  'close.approve.flow.view': 'Fold up the approval stream preview',
  'common.approve': 'The examination and approval',
  'common.pending.approval': 'Pending',
  'common.end': 'End',
  'workflow.preview': 'Workflow preview',
  'workflow.node': 'Node',
  'workflow.no.settings': 'No set up',
  'workflow.approval.chain': 'The examination and approval chain',
  'workflow.view.approval.chain': 'View approval chain',
  'workflow.dispatch': 'Scheduling',
  'workflow.scheduling.plan': 'Scheduling plan',
  'workflow.intelligence': 'Smart',
  'workflow.not.through': 'Do not pass through',
  'workflow.node.not.participate':
    'This object does not participate in the approval operation',
  'workflow.approval.record': 'The examination and approval records',
  'workflow.robot': 'The robot',
  'workflow.self.selected.node': 'Customize the node',
  'workflow.arrived': 'Have reached',
  'workflow.custom.approval': 'Optional approval',
  'company.maintain.company.companyLevelName': 'Company Level Name',
  'common.view': 'View',
  'base.sync.check.subsidiary.department':
    'Check the subordinate department at the same time',
  'common.simultaneous.approval': 'Simultaneous approval',
  'common.sequential.approval': 'Sequential approval',
  'peripheral.including.holidays': 'Including holiday',
  'common.before.modification': 'Before modification',
  'common.auditing.people': 'The reviewer',
  'workflow.financial.sharing': 'Financial Shared',
};
